src/array/cuda/spmm.cu

```diff
@@ -400,6 +501,109 @@ void SpMMCsr(const std::string& op, const std::string& reduce,
   }
 }
 
+/*!
+ * \brief CUDA implementation of g-SpMM on Csr format.
+ * \note use cusparse if the reduce operator is `sum` and there is
+ *       no broadcast, use dgl's kernel in other cases.
+ */
+template <int XPU, typename IdType, int bits>
+void SpMMCsrHetero(const std::string& op, const std::string& reduce,
+             const BcastOff& bcast,
+             const std::vector<CSRMatrix>& vec_csr,
+             std::vector<NDArray> vec_ufeat,
+             std::vector<NDArray> vec_efeat,
+             std::vector<NDArray> vec_out,
+             std::vector<NDArray> out_aux,
+             const std::vector<dgl_type_t> ufeat_eid,
+             const std::vector<dgl_type_t> out_eid) {
+  int64_t feat_len = bcast.out_len;
+  bool is_scalar_efeat = vec_efeat.size() != 0; //efeat.NumElements() == csr.indices->shape[0];
+  bool use_efeat = op != "copy_lhs";
+  const DLContext& ctx = DLContext{kDLGPU, 0};
```

Comment:
We should be getting the device from the input parameters (otherwise it won't work in a multi-GPU setting).